mixnet/src/node.rs

```diff
@@ -1,26 +1,118 @@
+use rand::rngs::OsRng;
 use serde::{Deserialize, Serialize};
 use tokio::sync::mpsc;
 
-use crate::{error::MixnetError, packet::Packet};
+use crate::fragment::{Fragment, MessageReconstructor};
+use crate::packet::{Message, Packet, PacketBody};
+use crate::{crypto::PrivateKey, error::MixnetError, poisson::Poisson};
 
 /// Mix node implementation that returns [`Output`] if exists.
-pub struct MixNode {}
+pub struct MixNode {
+    output_rx: mpsc::UnboundedReceiver<Output>,
+}
+
+struct MixNodeRunner {
+    config: MixNodeConfig,
+    poisson: Poisson,
+    packet_queue: mpsc::Receiver<Box<[u8]>>,
+    message_reconstructor: MessageReconstructor,
+    output_tx: mpsc::UnboundedSender<Output>,
+}
 
 /// Mix node configuration
 #[derive(Serialize, Deserialize, Clone, Debug)]
-pub struct MixNodeConfig {}
+pub struct MixNodeConfig {
+    /// Private key for decrypting Sphinx packets
+    pub encryption_private_key: PrivateKey,
+    /// Poisson delay rate per minutes
+    pub delay_rate_per_min: f64,
```

Comment:
Nitpick. Rates are usually normalized (`[0..1]` bounded). So we could use some specific type for it, but I am not sure if it actually exists as a crate.

Reply:
The rate can be bigger than 1 (`[0..inf]`) in Poisson distribution. In my opinion, it's better to call it frequency, but the rate is how it's called already: https://en.wikipedia.org/wiki/Poisson_distribution.

Reply:
Ah, its ok then.